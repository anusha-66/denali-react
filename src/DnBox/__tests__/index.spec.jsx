/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnBox } from '../../index'; // src/index.tsx

describe('DnBox Tests', () => {
    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBox>
                <h1>Title</h1>
            </DnBox>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBox className="p-18" style={{ width: '320px' }}>
                <h1>Title</h1>
            </DnBox>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with footer', () => {
        expect.assertions(1);

        const rendered = render(
            <DnBox className="p-18">
                <p>This is the body</p>
                <footer>This is a footer</footer>
            </DnBox>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
