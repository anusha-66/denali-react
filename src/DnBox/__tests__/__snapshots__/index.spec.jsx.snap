// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnBox Tests should render correctly with custom props 1`] = `
<div
  class="box p-18"
  style="width:320px"
>
  <h1>
    Title
  </h1>
</div>
`;

exports[`DnBox Tests should render correctly with footer 1`] = `
<div
  class="box p-18"
>
  <p>
    This is the body
  </p>
  <footer>
    This is a footer
  </footer>
</div>
`;

exports[`DnBox Tests should render with default props 1`] = `
<div
  class="box"
>
  <h1>
    Title
  </h1>
</div>
`;
